[iOS 16] Support building WebKit

emw-apple opened the pull request:
#### 7655272de94c4014c39c7c65178de952e25429fb
<pre>
[iOS 16] Support building WebKit
<a href="https://bugs.webkit.org/show_bug.cgi?id=244144">https://bugs.webkit.org/show_bug.cgi?id=244144</a>

Reviewed by Aditya Keerthi.

These changes are enough to get WebKit&apos;s frameworks building for iOS
Simulator on the public iOS 16 beta.

* Source/WTF/wtf/PlatformHave.h: Add HAVE() macro for UIBarButtonItem
  layout customization support.

* Source/WebCore/PAL/pal/spi/cocoa/NSAttributedStringSPI.h: NSTextList
  is newly public API. Use it instead of forward declarations.

* Source/WebCore/PAL/pal/spi/cocoa/PassKitSPI.h:
  HAVE(PASSKIT_MODULARIZATION) &amp;&amp; USE(APPLE_INTERNAL_SDK) is convention
  elsewhere in this file; I think it was accidentally missing here.

* Source/WebCore/PAL/pal/spi/cocoa/VisionKitCoreSPI.h: Add SPI
  declarations.
* Source/WebCore/PAL/pal/spi/ios/UIKitSPI.h: Ditto.
* Source/WebKit/Platform/spi/ios/UIKitSPI.h: Ditto.

* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h: Don&apos;t rely on
  the presence of a private header to determine whether we declare
  UIFindInteraction-related members.

* Source/WebKit/UIProcess/Cocoa/WKStorageAccessAlert.mm: Include SPI
  declarations.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj: Remove stale file
  reference, and commit sort order changes made by Xcode.

* Tools/TestWebKitAPI/ios/UIKitSPI.h: Add SPI declarations.

Remove stubs for iOS 13 and 14, add stubs for newly-linked-against
frameworks, and copy over frameworks we still link against in iOS 16.

* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/AssertionServices.framework/AssertionServices.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/CorePDF.framework/CorePDF.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/DeviceIdentity.framework/DeviceIdentity.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/AppSupport.framework/AppSupport.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/AssertionServices.framework/AssertionServices.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/AuthKit.framework/AuthKit.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/BackBoardServices.framework/BackBoardServices.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/CorePrediction.framework/CorePrediction.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/DeviceIdentity.framework/DeviceIdentity.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/GraphicsServices.framework/GraphicsServices.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/RunningBoardServices.framework/RunningBoardServices.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/URLFormatting.framework/URLFormatting.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/AppServerSupport.framework/AppServerSupport.tbd: Added.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/AppStoreDaemon.framework/AppStoreDaemon.tbd: Added.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/AppSupport.framework/AppSupport.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/AppSupport.framework/AppSupport.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/ApplePushService.framework/ApplePushService.tbd: Added.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/AuthKit.framework/AuthKit.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/AuthKit.framework/AuthKit.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/BackBoardServices.framework/BackBoardServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/BackBoardServices.framework/BackBoardServices.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/CorePrediction.framework/CorePrediction.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/CorePrediction.framework/CorePrediction.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/GraphicsServices.framework/GraphicsServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/GraphicsServices.framework/GraphicsServices.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/InstallCoordination.framework/InstallCoordination.tbd: Added.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/PrototypeTools.framework/PrototypeTools.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/PrototypeTools.framework/PrototypeTools.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/RunningBoardServices.framework/RunningBoardServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/RunningBoardServices.framework/RunningBoardServices.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/URLFormatting.framework/URLFormatting.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/URLFormatting.framework/URLFormatting.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/VisionKitCore.framework/VisionKitCore.tbd: Added.

Canonical link: <a href="https://commits.webkit.org/254391@main">https://commits.webkit.org/254391@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bd06cee7bce25400d0e5facda6392ecb5a5f5079

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/88771 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/33337 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/19658 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/97975 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/154487 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/92766 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/31842 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/27455 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/81015 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/92597 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/94401 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/25262 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/75758 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/25215 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/80139 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/80189 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/68178 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/80494 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/29640 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/14205 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/74275 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/29370 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/15204 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/26269 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3079 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/32808 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/38127 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/77136 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/31495 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34311 "Passed tests") | [❌ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/17097 "Found 1 new JSC stress test failure: stress/call-apply-exponential-bytecode-size.js.mini-mode (failure)") | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/WebKit/Platform/spi/ios/UIKitSPI.h:
```diff
@@ -1544,6 +1548,12 @@ typedef NS_ENUM(NSUInteger, _UIContextMenuLayout) {
 @property (nonatomic, setter=_setSuppressSoftwareKeyboard:) BOOL _suppressSoftwareKeyboard;
 @end
 
+#if HAVE(PASTEBOARD_DATA_OWNER) || HAVE(LINK_PREVIEW)
```
It feels like the proper condition should only be `HAVE(PASTEBOARD_DATA_OWNER)`, but _performAsDataOwner:block: gets called from: https://github.com/WebKit/WebKit/blob/da89ad902d18bbc6f8d03b805b8eaf859497fdf3/Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm#L11170
which is inside of a `ENABLE(IMAGE_ANALYSIS_ENHANCEMENTS)` condition. Above, we declare `_UIDataOwner` inside a `HAVE(LINK_PREVIEW)` condition.

Ideas for improving:

- We could move HAVE_PASTEBOARD_DATA_OWNER out of AdditionalFeatureDefines.h, and use it at the call site above
- We could remove the conditions around _UIDataOwner in this file, and declare these SPI more broadly

emw-apple replied:
After talking with @whsieh, I've done some cleanup around HAVE_PASTEBOARD_DATA_OWNER and split it out to #3726.

aproskuryakov commented on Source/WebCore/PAL/pal/spi/cocoa/NSAttributedStringSPI.h:
```diff
@@ -97,7 +97,7 @@ SOFT_LINK_CONSTANT(UIFoundation, NSTextAlternativesAttributeName, NSString *)
 
 #if HAVE(NSTEXTLIST_MARKER_FORMATS)
-#if USE(APPLE_INTERNAL_SDK)
+#if __has_include(<UIKit/NSTextList.h>)
```
One thing that I noticed while quickly scrolling through this patch: it's better to use OS version macros when possible, to make it obvious when obsolete code can be removed. An unnecessary `__has_include` branch can stay for years, as it's hard to find and hard to confirm that it's unneeded.

emw-apple commented on Source/WebKit/Platform/spi/ios/UIKitSPI.h:
```diff
@@ -1542,6 +1546,12 @@ typedef NS_ENUM(NSUInteger, _UIContextMenuLayout) {
+@interface UIPasteboard ()
++ (void)_performAsDataOwner:(_UIDataOwner)dataOwner block:(void(^)(void))block;
+@end
+#endif
```
This doesn't need to be guarded (it's available on all iOS versions we support), and HAVE(PASTEBOARD_DATA_OWNER) was cleaned up in #3726.

pxlcoder commented on Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
```diff
@@ -358,3 +358,9 @@ RetainPtr<NSError> nsErrorFromExceptionDetails(const WebCore::ExceptionDetails&)
+
+#if HAVE(UIFINDINTERACTION)
+@interface WKWebView (UIFindInteraction_Internal) <UITextSearching, UIFindInteractionDelegate>
+@property (nonatomic, readwrite, setter=_setFindInteractionEnabled:) BOOL _findInteractionEnabled;
```
Why is this declaration necessary in the Internal header? (the same declaration is in the Private header)

pxlcoder commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPage.mm:
```diff
@@ -334,7 +334,14 @@ static NSUInteger replaceMatches(WKWebView *webView, NSArray<FindMatch> *matches
+
+- (void)_addLayerForFindOverlay;
+- (void)_removeLayerForFindOverlay;
+- (void)_requestRectForFoundTextRange:(UITextRange *)ranges completionHandler:(void (^)(CGRect))completionHandler;
```
Can we just include `WKWebViewPrivate.h` to pull these in? (slightly surprised that isn't already the case)

emw-apple replied:
We do already include WKWebViewPrivate.h. The challenge is that it only declares these when the private SDK is available:
```
#if __has_include(<UIKit/_UIFindInteraction.h>)
@property (nonatomic, readonly) _UIFindInteraction *_findInteraction WK_API_AVAILABLE(ios(16.0));
@property (nonatomic, readwrite, setter=_setFindInteractionEnabled:) BOOL _findInteractionEnabled WK_API_AVAILABLE(ios(16.0));

@property (nonatomic, readonly) CALayer *_layerForFindOverlay WK_API_AVAILABLE(ios(16.0));

- (void)_requestRectForFoundTextRange:(UITextRange *)ranges completionHandler:(void (^)(CGRect))completionHandler WK_API_AVAILABLE(ios(16.0));

- (void)_addLayerForFindOverlay WK_API_AVAILABLE(ios(16.0));
- (void)_removeLayerForFindOverlay WK_API_AVAILABLE(ios(16.0));
#endif
```

So we can't rely solely on WKWebViewPrivate.h without emitting a bunch of redeclarations in our private headers.

I spent some time trying to move usage over to `UIFindInteraction` (rdar://95125552), which is new API in iOS 16. I was able to clean things up nicely and avoid these redeclarations, but that broke the open source iOS 15 build. I could spend a bit more time trying to get that working, but it might be better to wait until next year when we drop support for building with the iOS 15 SDK.

pxlcoder replied:
I see!

We should just be able to change the `__has_include` to `TARGET_OS_IOS && __IPHONE_OS_VERSION_MIN_REQUIRED >= 160000`.

emw-apple replied:
Our style checker doesn't like `__IPHONE_OS_VERSION_MIN_REQUIRED` in private headers:

    ERROR: Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:500:  Misplaced OS version check. Please use a named macro in one of headers in the wtf/Platform.h suite of files or an appropriate internal file.  [build/version_check] [5]

I spent some more time messing with UIFindInteraction, and was able to clean up our remaining usage of `_UITextSearching` and `_UIFindInteraction` SPI in https://github.com/WebKit/WebKit/pull/3494/commits/98a38143d27572483303206082f81850cfe70954. That, plus switching to checking for the public header (`__has_include(<UIKit/UIFindInteraction.h>)`) seems to be working well.

pxlcoder replied:
I think we can ignore that style warning. We can't use the platform macros in this file, so I don't think the warning applies here.

We cannot remove the SPI since we have clients that still rely on it.

emw-apple replied:
Got it! PR updated with your feedback, here's a diff: https://github.com/WebKit/WebKit/compare/9c5d061133b9eb82a7593d8e7b605a377a37562c..7e2ce29be192a73eedba1d402920529b0a5fb649

pxlcoder commented on Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
```diff
@@ -120,6 +120,7 @@ typedef NS_OPTIONS(NSUInteger, _WKRectEdge) {
 @class WKBrowsingContextHandle;
 @class WKDownload;
 @class WKFrameInfo;
+@class _UIFindInteraction;
```
Nit: Can we move this up underneath `@class UIEventAttribution;`?

aproskuryakov replied:
We typically sort everything in ASCII order, respecting underscores and such.

pxlcoder replied:
In that case, we should move this after `@class WKWebpagePreferences;`.

pxlcoder reviewed (approved):
`UIFindInteraction` changes look good!

emw-apple committed:
[iOS 16] Support building WebKit
https://bugs.webkit.org/show_bug.cgi?id=244144

Reviewed by Aditya Keerthi.

These changes are enough to get WebKit's frameworks building for iOS
Simulator on the public iOS 16 beta.

* Source/WTF/wtf/PlatformHave.h: Add HAVE() macro for UIBarButtonItem
  layout customization support.

* Source/WebCore/PAL/pal/spi/cocoa/NSAttributedStringSPI.h: NSTextList
  is newly public API. Use it instead of forward declarations.

* Source/WebCore/PAL/pal/spi/cocoa/PassKitSPI.h:
  HAVE(PASSKIT_MODULARIZATION) && USE(APPLE_INTERNAL_SDK) is convention
  elsewhere in this file; I think it was accidentally missing here.

* Source/WebCore/PAL/pal/spi/cocoa/VisionKitCoreSPI.h: Add SPI
  declarations.
* Source/WebCore/PAL/pal/spi/ios/UIKitSPI.h: Ditto.
* Source/WebKit/Platform/spi/ios/UIKitSPI.h: Ditto.

* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h: Don't rely on
  the presence of a private header to determine whether we declare
  UIFindInteraction-related members.

* Source/WebKit/UIProcess/Cocoa/WKStorageAccessAlert.mm: Include SPI
  declarations.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj: Remove stale file
  reference, and commit sort order changes made by Xcode.

* Tools/TestWebKitAPI/ios/UIKitSPI.h: Add SPI declarations.

Remove stubs for iOS 13 and 14, add stubs for newly-linked-against
frameworks, and copy over frameworks we still link against in iOS 16.

* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/AssertionServices.framework/AssertionServices.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/CorePDF.framework/CorePDF.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/DeviceIdentity.framework/DeviceIdentity.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/AppSupport.framework/AppSupport.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/AssertionServices.framework/AssertionServices.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/AuthKit.framework/AuthKit.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/BackBoardServices.framework/BackBoardServices.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/CorePrediction.framework/CorePrediction.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/DeviceIdentity.framework/DeviceIdentity.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/GraphicsServices.framework/GraphicsServices.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/RunningBoardServices.framework/RunningBoardServices.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/URLFormatting.framework/URLFormatting.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/AppServerSupport.framework/AppServerSupport.tbd: Added.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/AppStoreDaemon.framework/AppStoreDaemon.tbd: Added.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/AppSupport.framework/AppSupport.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/AppSupport.framework/AppSupport.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/ApplePushService.framework/ApplePushService.tbd: Added.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/AuthKit.framework/AuthKit.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/AuthKit.framework/AuthKit.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/BackBoardServices.framework/BackBoardServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/BackBoardServices.framework/BackBoardServices.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/CorePrediction.framework/CorePrediction.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/CorePrediction.framework/CorePrediction.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/GraphicsServices.framework/GraphicsServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/GraphicsServices.framework/GraphicsServices.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/InstallCoordination.framework/InstallCoordination.tbd: Added.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/PrototypeTools.framework/PrototypeTools.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/14/PrototypeTools.framework/PrototypeTools.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/RunningBoardServices.framework/RunningBoardServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/RunningBoardServices.framework/RunningBoardServices.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/URLFormatting.framework/URLFormatting.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/13/URLFormatting.framework/URLFormatting.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/VisionKitCore.framework/VisionKitCore.tbd: Added.

Canonical link: https://commits.webkit.org/254391@main
